### PR TITLE
docs: change assert to raise and some small docs

### PR DESCRIPTION
**Pull request by pphuc25**
Hi,
I create a concise document outlining the process of improving code readability by replacing 'assert' statements with 'raise' statements, along with the addition of documentation and 'logger.info' statements.
I would like cc @stevhliu to review my PR.

### CODE REVIEW

**Comment by stevhliu** [comment]
Hi, in general we like to use `raise` statements instead of `assert` because the latter are considered more like debugging statements that can be toggled on or off in certain setups.

**Comment by pphuc25** [comment]
So to catch more debugging information, I switch from using assert to raise. I believe the code I am working on is correct, @stevhliu.

**Review by stevhliu — COMMENTED** [review]
Sorry I read it the wrong way, you're correct! The rest of the changes look good to me. 🙂 

Can you try running `make style` to fix the failing CI tests?

**Comment by pphuc25** [comment]
Hi @stevhliu, can you help me, I cannot run this command

**Comment by stevhliu** [comment]
Yes, it looks like you're missing `black` which can be fixed by installing it first:

```bash
pip install -e ".[quality]"
```

Then you should be able to run the `make style` command :)

**Review by stevhliu — APPROVED** [review]
Thanks @pphuc25! Pinging @LysandreJik for a final look :)

**Review by LysandreJik — COMMENTED** [review]
Thank you for your changes @pphuc25, but let's focus on your changes from assert to raise statements for now.

Please revert the added comments/changes in logging statements; we want these scripts to be very minimal so we'll keep them the way they are currently for now.

Thank you!

**Comment by pphuc25** [comment]
Hi @LysandreJik, I have reverted them back. However, in run_mlm_no_trainer the logging do not have eval_loss, this seem the important one so user can know how well loss of their model is, I think we should keep add the eval_loss logging in that file.

File: examples/pytorch/language-modeling/run_clm_no_trainer.py [hunk 1]
```diff
@@ -487,9 +490,11 @@ def group_texts(examples):
     train_dataset = lm_datasets["train"]
     eval_dataset = lm_datasets["validation"]
 
-    # Log a few random samples from the training set:
-    for index in random.sample(range(len(train_dataset)), 3):
-        logger.info(f"Sample {index} of the training set: {train_dataset[index]}.")
+    # Conditional for small test subsets
+    if len(train_dataset) > 3:
```
Comment (LysandreJik) [inline examples/pytorch/language-modeling/run_clm_no_trainer.py:494]: I would revert this change

File: examples/pytorch/language-modeling/run_clm_no_trainer.py [hunk 2]
```diff
@@ -514,6 +519,9 @@ def group_texts(examples):
     ]
     optimizer = torch.optim.AdamW(optimizer_grouped_parameters, lr=args.learning_rate)
 
+    # Note -> the training dataloader needs to be prepared before we grab his length below (cause its length will be
+    # shorter in multiprocess)
+
```
Comment (LysandreJik) [inline examples/pytorch/language-modeling/run_clm_no_trainer.py:524]: same here

File: examples/pytorch/language-modeling/run_clm_no_trainer.py [hunk 3]
```diff
@@ -636,6 +644,7 @@ def group_texts(examples):
                     if args.output_dir is not None:
                         output_dir = os.path.join(args.output_dir, output_dir)
                     accelerator.save_state(output_dir)
+
```
Comment (LysandreJik) [inline examples/pytorch/language-modeling/run_clm_no_trainer.py:647]: same here

File: src/transformers/models/persimmon/modeling_persimmon.py [hunk 4]
```diff
@@ -72,7 +72,7 @@ def _expand_mask(mask: torch.Tensor, dtype: torch.dtype, tgt_len: Optional[int]
 
 
 # Copied from transformers.models.llama.modeling_llama.LlamaRotaryEmbedding with Llama->Persimmon
-class PersimmonRotaryEmbedding(torch.nn.Module):
+class PersimmonRotaryEmbedding(nn.Module):
```
Comment (LysandreJik) [inline src/transformers/models/persimmon/modeling_persimmon.py:75]: same here

**Review by LysandreJik — COMMENTED** [review]
I have written down the changes that should be reverted in my opinion, the rest (including the eval loss), can be kept!

Thanks!

File: src/transformers/models/persimmon/modeling_persimmon.py (hunk 4 above)
Comment (pphuc25) [reply src/transformers/models/persimmon/modeling_persimmon.py:75]: This is the code after running `make fix-copies`, this change because transformers in the past have some bug, so this is the automatic code to pass the test on github, we should keep it.

Comment (LysandreJik) [reply src/transformers/models/persimmon/modeling_persimmon.py:75]: Sounds good, thanks @pphuc25

**Review by LysandreJik — APPROVED** [review]
Thank you for your PR!
